grapher/utils/Util.ts

```diff
@@ -1045,3 +1068,86 @@ export const splitArrayIntoGroupsOfN = (arr: any[], maxPerGroup: number) => {
         result.push(arr.slice(index, index + maxPerGroup))
     return result
 }
+
+export function getClosestTimePairs(
+    sortedTimesA: Time[],
+    sortedTimesB: Time[],
+    maxDiff: Integer = Infinity
+) {
+    if (sortedTimesA.length === 0 || sortedTimesB.length === 0) return []
+
+    const decidedPairs: [Time, Time][] = []
+    const undecidedPairs: [Time, Time][] = []
+
+    let indexB = 0
+
+    for (let indexA = 0; indexA < sortedTimesA.length; indexA++) {
+        const timeA = sortedTimesA[indexA]
+
+        const closestIndexInB = sortedFindClosestIndex(
+            sortedTimesB,
+            timeA,
+            indexB
+        )
+
+        /**
+         * the index that holds the value that is definitely lower than timeA, the candidate time
+         */
+        const lowCandidateIndexB =
+            sortedTimesB[closestIndexInB] < timeA
+                ? closestIndexInB
+                : closestIndexInB > indexB
+                ? closestIndexInB - 1
+                : undefined
+
+        /**
+         * the index that holds the value that is definitely equal to or greater than timeA, the candidate time
+         */
+        const highCandidateIndexB =
+            sortedTimesB[closestIndexInB] >= timeA ? closestIndexInB : undefined
+
+        if (
+            lowCandidateIndexB !== undefined &&
+            highCandidateIndexB !== undefined &&
+            timeA - sortedTimesB[lowCandidateIndexB] <= maxDiff &&
+            timeA - sortedTimesB[lowCandidateIndexB] <
+                sortedTimesB[highCandidateIndexB] - timeA
+        ) {
```

Comment:
_super_ nit: maybe drop the {}? I prefer no "{}" when the body of a block is only one line. It tells me that the block is as simple as can be. but suppper nit

Reply:
(obviously no need to fix any of these nits before merignG)

Reply:
I think no `{}` works well when the full statement is in a single line, but otherwise find `{}` better. 

In a multi-line, you might add a line above/below and it would compile without errors